adapter/src/utils/useLocale.js

```diff
@@ -47,16 +95,21 @@ const settingsQuery = {
         resource: 'userSettings',
     },
 }
-export const useCurrentUserLocale = () => {
+// note: userSettings.keyUiLocale is expected to be in the Java format,
+// e.g. 'ar', 'ar_IQ', 'uz_UZ_Cyrl', etc.
+export const useCurrentUserLocale = (configDirection) => {
     const { loading, error, data } = useDataQuery(settingsQuery)
-    const locale = useLocale(
-        data && (data.userSettings.keyUiLocale || window.navigator.language)
-    )
+    const { locale, direction } = useLocale({
+        locale:
+            data &&
+            (data.userSettings.keyUiLocale || window.navigator.language),
+        configDirection,
+    })
 
     if (error) {
         // This shouldn't happen, trigger the fatal error boundary
         throw new Error('Failed to fetch user locale: ' + error)
     }
 
-    return { loading: loading || !locale, locale }
+    return { loading: loading || !locale, locale, direction }
```

Comment:
it would be nice to add a test for this hook .. I know it's not particularly part of this PR since it didn't exist before, but there is quite a bit of logic here and having a test to validate all the permuations would be nice

Reply:
That may take me some time since I'm a bit out of practice with mocking things like `useDataQuery` and `i18n.dir`, but I'll see what I can do 👍

Reply:
ok, we can delay it for later too .. but I didn't mean to test the whole thing, just the useLocale hook with something like [react-hooks-testing-library](https://github.com/testing-library/react-hooks-testing-library)

Reply:
Ah okay, just the `useLocale` hook will be quite a bit simpler -- this comment was placed in a different hook though 😁

Reply:
Tests added